source/collections/Vector2D.ooc

```diff
@@ -8,82 +8,29 @@
 
 Vector2D: class <T> {
 	_backend: T*
-	_rowCount: Int
-	_columnCount: Int
-	rowCount ::= this _rowCount
-	columnCount ::= this _columnCount
-
-	init: func ~preallocated (=_backend, =_rowCount, =_columnCount)
-	init: func (=_rowCount, =_columnCount) {
-		this _allocate(this rowCount, this columnCount)
-		memset(this _backend, 0, this rowCount * this columnCount * T size)
+	_width: Int
+	_height: Int
+	width ::= this _width
+	height ::= this _height
+	init: func ~preallocated (=_backend, =_width, =_height)
+	init: func (=_width, =_height) {
+		this _backend = calloc(this _width * this _height, T size)
 	}
 	free: override func {
 		memfree(this _backend)
 		super()
 	}
-	_allocate: func (rows, columns: Int) {
-		this _backend = realloc(this _backend, rows * columns * T size)
-	}
-	_elementPosition: func (row, column: Int, columnCount := this columnCount) -> Int {
-		columnCount * row + column
-	}
-	resize: func (newRowCount, newColumnCount: Int) {
-		if (newColumnCount == this columnCount) {
-			if (newRowCount != this rowCount) {
-				this _allocate(newRowCount, newColumnCount)
-				this _rowCount = newRowCount
-				this _columnCount = newColumnCount
-			}
-		} else {
-			minimumRowCount := this rowCount minimum(newRowCount)
-			minimumColumnCount := this columnCount minimum(newColumnCount)
-			temporaryResult: T* = calloc(newRowCount * newColumnCount, T size)
-
-			for (row in 0 .. minimumRowCount)
-				memcpy(temporaryResult[T size * this _elementPosition(row, 0, newColumnCount)]&,
-					this _backend[T size * this _elementPosition(row, 0)]&, minimumColumnCount * T size)
-
-			memfree(this _backend)
-			this init(temporaryResult, newRowCount, newColumnCount)
-		}
-	}
-	move: func (sourceRowStart, sourceColumnStart, targetRowStart, targetColumnStart: Int, columnCount := 0, rowCount := 0) {
-		sourceRowIndex, targetRowIndex: Int
-
-		if (rowCount < 1)
-			rowCount = this rowCount - sourceRowStart
-		if (columnCount < 1)
-			columnCount = this columnCount - sourceColumnStart
-		if (targetRowStart + rowCount > this rowCount)
-			rowCount = this rowCount - targetRowStart
-		if (targetColumnStart + columnCount > this columnCount)
-			columnCount = this columnCount - targetColumnStart
-
-		for (row in 0 .. rowCount) {
-			if (sourceRowStart > targetRowStart) {
-				// Move row-wise from top to bottom.
-				sourceRowIndex = sourceRowStart + row
-				targetRowIndex = sourceRowIndex - sourceRowStart + targetRowStart
-			} else {
-				//Move row-wise from bottom to top
-				targetRowIndex = rowCount + targetRowStart - row - 1
-				sourceRowIndex = targetRowIndex - targetRowStart + sourceRowStart
-			}
-
-			memmove(this _backend[T size * this _elementPosition(targetRowIndex, targetColumnStart)]&,
-				this _backend[T size * this _elementPosition(sourceRowIndex, sourceColumnStart)]&, columnCount * T size)
-		}
+	_elementPosition: func (x, y: Int, width := this width) -> Int {
+		width * y + x
```

Comment:
I'm curious about this. First of all, the third argument is superfluous, it's always the default.
Second, it's only used in `[]` and `[]=` operators, we should just inline it there.